### PR TITLE
Check x and y after preprocessing

### PR DESCRIPTION
**Pull request by jan-gerling**
* check and assert x and y after preprocessing
 * convert x and y to np arrays instead of dataframes

### CODE REVIEW

File: ml/preprocessing/preprocessing.py [hunk 1]
```diff
@@ -123,5 +123,11 @@ def retrieve_labelled_instances(dataset, refactoring: LowLevelRefactoring, is_tr
...
         x = perform_scaling(x, scaler)
 
+    # finally validate x and y
+    features = x.columns.values
+    x, y = check_X_y(x, y)
+    assert x.shape == (y.shape[0], len(features)), f"X has the wrong shape after checking: {x.shape}"
+    assert y.shape[0] == x.shape[0], f"Y has the wrong shape after checking: {y.shape}"
+        
     log("Got %d instances with %d features for the dataset: %s." % (x.shape[0], x.shape[1], dataset))
     return x.columns.values, x, y, ids, scaler
```
Comment (Dahny) [inline ml/preprocessing/preprocessing.py]: x.columns.values -> should be features now as x is not a dataframe anymore

**Review by Dahny — CHANGES_REQUESTED** [review]
The binary.py still requires the pandas import in the evaluate_model

File: ml/preprocessing/preprocessing.py (hunk 1 above)
Comment (jan-gerling) [reply ml/preprocessing/preprocessing.py]: I fixed the merge error.

**Comment by jan-gerling** [comment]
I will close this PR and discard the changes, because with a numpy array we also loose the column names and thus, the analysis of the feature relevance becomes more difficult. Also, the preprocessing is well tested now.

**Pull request closed without merging**